File: src/main/java/mk/ukim/finki/wp/emtlab/model/domain/Accomodation.java
package mk.ukim.finki.wp.emtlab.model.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import mk.ukim.finki.wp.emtlab.model.enums.Category;
import mk.ukim.finki.wp.emtlab.model.enums.Status;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "accomodations")
public class Accomodation extends BaseAuditableEntity {
    @Column(nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    private Category category;

    @ManyToOne
    @JoinColumn(name = "host_id", nullable = false)
    private Host host;

    @Enumerated(EnumType.STRING)
    private Status status;

    @Column(nullable = false)
    private Integer numRooms;

    @Column(nullable = false)
    private Boolean rented;

    public Accomodation(String name, Category category, Host host, Status status, Integer numRooms, Boolean rented) {
        this.name = name;
        this.category = category;
        this.host = host;
        this.status = status;
        this.numRooms = numRooms;
        this.rented = rented;
    }
}

File: frontend/src/types/accomodation.ts
export type Category = 'APARTMENT' | 'HOUSE' | 'MOTEL' | 'HOTEL' | 'FLAT' | 'ROOM';
export type Status = 'GOOD' | 'BAD' | 'NEW' | 'RENOVATED';

export interface Accomodation {
  id: number;
  name: string;
  category: Category;
  status: Status;
  numRooms: number;
  hostId: number;
  rented: boolean;
}

